pkg/k8s/client.go

```diff
@@ -122,6 +123,17 @@ func setDialer(config *rest.Config) func() {
 }
 
 func runHeartbeat(heartBeat func(context.Context) error, timeout time.Duration, closeAllConns ...func()) {
+	expireDate := time.Now().Add(-timeout)
+	// Don't even perform a health check if we have received a k8s event
+	// in the last 'timeout' duration
+	if k8smetrics.LastInteraction.Time().After(expireDate) {
+		// However, if we have seen an error in the last
+		// 'timeout' duration then we should close all connections!
```

Comment:
The comment seems wrong. It's not what the code does. The comment should say:
```
// Skip the heartbeat if we have received an event within the heartbeat timeout interval and not error has been received from the apiserver.
```

That said, is the error condition really useful. We are likely to receive quite a few errors while attempting to update outdated resources. Would they all skip the heartbeat ping?